op-node/rollup/derive/channel_compressor.go

```diff
@@ -0,0 +1,85 @@
+package derive
+
+import (
+	"bytes"
+	"compress/zlib"
+	"fmt"
+	"io"
+
+	"github.com/andybalholm/brotli"
+)
+
+const (
+	ChannelVersionBrotli byte = 0x01
+)
+
+type CompressorWriter interface {
+	Write([]byte) (int, error)
+	Flush() error
+	Close() error
+	Reset(io.Writer)
+}
+
+type ChannelCompressor struct {
+	writer          CompressorWriter
+	compressionAlgo CompressionAlgo
+	compressed      *bytes.Buffer
+}
+
+func NewChannelCompressor(algo CompressionAlgo) (*ChannelCompressor, error) {
+	var writer CompressorWriter
+	var err error
+	compressed := &bytes.Buffer{}
+	if algo == Zlib {
+		writer, err = zlib.NewWriterLevel(compressed, zlib.BestCompression)
+	} else if algo.IsBrotli() {
+		compressed.WriteByte(ChannelVersionBrotli)
```

Comment:
could just call Reset on the compressor, which would also write this byte for you